pkg/cloudprovider/aws/cloudprovider.go

```diff
@@ -80,14 +79,8 @@ func NewCloudProvider(ctx context.Context, options cloudprovider.Options) *Cloud
 	return &CloudProvider{
 		instanceTypeProvider: instanceTypeProvider,
 		subnetProvider:       subnetProvider,
-		instanceProvider: &InstanceProvider{ec2api, instanceTypeProvider, subnetProvider,
-			NewLaunchTemplateProvider(
-				ec2api,
-				NewAMIProvider(ssm.New(sess), options.ClientSet),
-				NewSecurityGroupProvider(ec2api),
-			),
-		},
-		creationQueue: parallel.NewWorkQueue(CreationQPS, CreationBurst),
+		instanceProvider:     NewInstanceProvider(ec2api, instanceTypeProvider, subnetProvider, sess, options.ClientSet),
```

Comment:
I love that you've encapsulated the LT provider inside the instance provider. Do you mind instantiating an SSM client out here and passing it into NewInstanceProvider instead of sess? This reduces the scope of what lower level constructors are able to instantiate and makes it more clear what services the cloud provider relies on by just looking at this construct.